### PR TITLE
UINV-419 Allow user to process invoices against previous fiscal years

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://issues.folio.org/browse/UINV-419

Add the ability for a permitted user to select fiscal year when creating/editing an invoice.
## Approach

- Introduce new permission `Invoice: Pay invoices in a different fiscal year`;
- Implement form field component to select an FY;
- Implement key-value component to display value on invoice details;
- Implement hooks to fetch payable fiscal years (current and previous) - `usePayableFiscalYears` and to fetch fiscal year by ID - `useFiscalYear`;
- Cover new functionality with unit tests.

<details open><summary>Preview</summary>
<p>

chrome_ZDXH32Rc2C

</p>
</details> 

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/common/hooks/usePayableFiscalYears/usePayableFiscalYears.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { FISCAL_YEARS_API } from '../../constants';
+
+const INITIAL_DATA = [];
+
+export const usePayableFiscalYears = (options = {}) => {
+  const intl = useIntl();
+  const ky = useOkapiKy();
+  const [namespace] = useNamespace({ key: 'invoice-payable-fiscal-years' });
+
+  const periodStart = new Date(intl.formatDate(new Date(), { timeZone: TIMEZONE })).toISOString();
```
Comment (NikitaSedyx) [inline src/common/hooks/usePayableFiscalYears/usePayableFiscalYears.js:22]: TIMEZONE is utc, isn't?

**Review by NikitaSedyx — APPROVED** [review]
with assumption that timezone is utc

File: src/common/hooks/usePayableFiscalYears/usePayableFiscalYears.js (hunk 1 above)
Comment (usavkov-epam) [reply src/common/hooks/usePayableFiscalYears/usePayableFiscalYears.js]: yep, you're right

File: src/common/components/FiscalYearValue/FiscalYearValueContainer.test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { render, screen } from '@testing-library/react';
+
+import { FiscalYearValueContainer } from './FiscalYearValueContainer';
+
+jest.mock('../../hooks', () => ({
+  useFiscalYear: jest.fn(() => ({
```
Comment (mariia-aloshyna) [inline src/common/components/FiscalYearValue/FiscalYearValueContainer.test.js:6]:
```suggestion
jest.mock('../../hooks', () => ({
  ...jest.requireActual('../../hooks'),
  useFiscalYear: jest.fn(() => ({
```

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**